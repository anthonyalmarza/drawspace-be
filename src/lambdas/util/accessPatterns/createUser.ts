import userEntity from 'lambdas/util/entities/user'
import { User, UpdateReturnValues } from 'types'

export default async ({
    userId,
    name,
}: {
    userId: string
    name: string
}): Promise<User> => {
    const { Attributes: user } = await userEntity.update(
        { id: userId, name },
        {
            returnValues: UpdateReturnValues.ALL_NEW,
            conditions: { attr: 'pk', exists: false },
        }
    )
    return user
}
