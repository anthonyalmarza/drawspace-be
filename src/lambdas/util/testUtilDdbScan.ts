import DynamoDB from 'aws-sdk/clients/dynamodb'

const { apiTableName } = require('lambdas/util/envVarsRestApi')
/* eslint-enable */
const DocumentClient = new DynamoDB.DocumentClient({
    endpoint: 'http://localhost:9001',
    accessKeyId: 'dynamo',
    secretAccessKey: 'devDummyKey',
    region: 'local',
})

export default () => DocumentClient.scan({ TableName: apiTableName }).promise()
