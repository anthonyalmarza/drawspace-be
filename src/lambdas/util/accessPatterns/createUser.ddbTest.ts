import { advanceTo } from 'jest-date-mock'
import testUtilDdbScan from 'lambdas/util/testUtilDdbScan'

import createUser from 'lambdas/util/accessPatterns/createUser'

advanceTo(new Date(2020, 5, 29, 0, 0, 0))

const mockUserId = '1234123412341234'

test('createUser', async () => {
    const res = await createUser({ userId: mockUserId, name: 'mock user name' })
    const scan = await testUtilDdbScan()
    expect(res).toMatchSnapshot()
    expect(scan).toMatchSnapshot()
})
