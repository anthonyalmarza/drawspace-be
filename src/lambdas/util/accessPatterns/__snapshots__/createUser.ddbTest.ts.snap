// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createUser 1`] = `
Object {
  "created": "2020-06-29T00:00:00.000Z",
  "entity": "user",
  "id": "1234123412341234",
  "modified": "2020-06-29T00:00:00.000Z",
  "name": "mock user name",
  "privateDrawingCount": 0,
  "publicDrawingCount": 0,
}
`;

exports[`createUser 2`] = `
Object {
  "Count": 1,
  "Items": Array [
    Object {
      "_ct": "2020-06-29T00:00:00.000Z",
      "_et": "user",
      "_md": "2020-06-29T00:00:00.000Z",
      "dc": 0,
      "n": "mock user name",
      "pdc": 0,
      "pk": "user#1234123412341234",
      "sk": "user#1234123412341234",
    },
  ],
  "ScannedCount": 1,
}
`;
